### PR TITLE
Deprecate cirq.ion module

### PR DESCRIPTION
**Pull request by tanujkhattar**
Fixes https://github.com/quantumlib/Cirq/issues/2793
Fixes https://github.com/quantumlib/Cirq/issues/5128

### CODE REVIEW

File: cirq-core/cirq/ion/ion_device_test.py [hunk 1]
```diff
@@ -19,16 +19,20 @@
 import cirq
 import cirq.ion as ci
 import cirq.testing
+import sympy.utilities.matchpy_connector
```
Comment (MichaelBroughton) [inline cirq-core/cirq/ion/ion_device_test.py:22]: Is this needed ?

**Review by MichaelBroughton — APPROVED** [review]
LGTM, do we want to deprecate `cirq.ion` like we did with `cirq.google` or does getting all the major things inside of it seem like enough ?

File: cirq-core/cirq/ion/ion_device_test.py (hunk 1 above)
Comment (tanujkhattar) [reply cirq-core/cirq/ion/ion_device_test.py:22]: Nope, replaced with `import sympy` instead.

**Comment by tanujkhattar** [comment]
> do we want to deprecate cirq.ion like we did with cirq.google or does getting all the major things inside of it seem like enough

Since we have deprecated everything inside the module, we don't need to deprecate the `cirq.ion` submodule itself. The latter is needed when we are moving utilities but want the users to update the path (for example, use `cirq_google.Sycamore` instead of `cirq.google.Syamore`). In this case, no object inside the `cirq.ion` is preserved -- everything is replaced by newer alternatives and therefore just deprecating all the objects is enough.

**Pull request merged**